contracts/src/xYield/xYieldVault.sol

```diff
@@ -116,6 +123,12 @@ contract xYieldVault is ERC4626, Ownable2Step, ReentrancyGuard, Pausable {
         _depositFrom(msg.sender, _receiver, _amount, shares, _chainId);
     }
 
+    function reDepositIdleAssets() external {
```

Comment:
we don't want this, but especially without a modifier to restrict permissions. there may be reasons why we don't want to reinvest idle assets, such as funds temporarily being available for remote withdrawal by a user. anyway we have a logic determine to only invest funds if the chain is active, and this breaks that logic.